extensions/common/api/api-core/src/main/java/org/eclipse/edc/api/transformer/JsonObjectFromCriterionDtoTransformer.java

```diff
@@ -0,0 +1,52 @@
+/*
+ *  Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.api.transformer;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.json.JsonBuilderFactory;
+import jakarta.json.JsonObject;
+import jakarta.json.JsonValue;
+import org.eclipse.edc.api.model.CriterionDto;
+import org.eclipse.edc.jsonld.spi.transformer.AbstractJsonLdTransformer;
+import org.eclipse.edc.transform.spi.TransformerContext;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import static org.eclipse.edc.api.model.CriterionDto.CRITERION_OPERAND_LEFT;
+import static org.eclipse.edc.api.model.CriterionDto.CRITERION_OPERAND_RIGHT;
+import static org.eclipse.edc.api.model.CriterionDto.CRITERION_OPERATOR;
+import static org.eclipse.edc.jsonld.spi.JsonLdKeywords.TYPE;
+import static org.eclipse.edc.spi.CoreConstants.EDC_NAMESPACE;
+
+public class JsonObjectFromCriterionDtoTransformer extends AbstractJsonLdTransformer<CriterionDto, JsonObject> {
+    private final JsonBuilderFactory jsonFactory;
+    private final ObjectMapper mapper;
+
+    public JsonObjectFromCriterionDtoTransformer(JsonBuilderFactory jsonFactory, ObjectMapper mapper) {
+        super(CriterionDto.class, JsonObject.class);
+        this.jsonFactory = jsonFactory;
+        this.mapper = mapper;
+    }
+
+    @Override
+    public @Nullable JsonObject transform(@NotNull CriterionDto criterionDto, @NotNull TransformerContext context) {
+        var builder = jsonFactory.createObjectBuilder();
+        builder.add(TYPE, EDC_NAMESPACE + "CriterionDto")
```

Comment:
constant `CRITERION_TYPE` from `CriterionDto` can be used here

Reply:
yep, must have slipped through...